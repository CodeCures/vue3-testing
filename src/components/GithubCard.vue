<script setup lang="ts">
import { ref } from 'vue';

    const user = ref(null);
    const username = ref('');
    const url = ref(null)

    function composeUrl (userName: string){
        return url.value = `https://api.github.com/users/${userName}`;
    }

    const getUser = async () => {
        const response = await fetch(url.value);
        user.value = response.json();
        
    }
    
</script>

<template>
    <div v-if="user">
        <img :src="user.avatar_url" :alt="user.name">
        <h1>{{ user.name }}</h1>
        <p>{{ user.bio }} -- {{ user.location }}</p>
    </div>
</template>