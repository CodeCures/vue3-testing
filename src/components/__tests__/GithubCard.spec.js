import { beforeEach, describe, expect, test, vi } from "vitest";
import GithubCard from '@/components/GithubCard.vue'
import { mount } from "@vue/test-utils";

describe("methods", () => {
    let wrapper;
    let vm;

    global.fetch = vi.fn();

    beforeEach(() => {
        if (wrapper) wrapper.unmount();

        wrapper = mount(GithubCard);
        vm = wrapper.vm
    })

    test("composeUrl", () => {
        expect(vm.composeUrl('123')).toBe("https://api.github.com/users/123")
        expect(vm.composeUrl('codeCures')).toBe("https://api.github.com/users/codeCures")
    });

    test("fetUser", async () => {

        const gitUser = (await import('./utils/github_user.json')).default;

        fetch.mockResolvedValue({
            json: () => gitUser
        })

        vm.composeUrl('codeCures')
        await vm.getUser()


        expect(fetch).toHaveBeenCalledWith('https://api.github.com/users/codeCures')
        expect(vm.user).toStrictEqual(gitUser)
        expect(wrapper.text()).toMatchSnapshot()
        
    })


})