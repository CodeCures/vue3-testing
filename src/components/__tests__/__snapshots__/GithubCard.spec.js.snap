// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`methods > fetUser 1`] = `"CourageChidi Courage Francis. A Senior Software Developer at codesGate. With over 8 years experience in software development as a full stack developer -- Lagos, Nigeria"`;
